### PR TITLE
Newsletter/form validation

### LINKED ISSUE #121: [Bug] Poor Form Validation in newsletter signup form
## Description
When submitting the subscribe to the newsletter form in the footer section, there's a redirection to a page to provide additional information. However, if there are validation errors (e.g., invalid email), the fields with errors do not display any visible validation messages or indicators for the fields affected. Although the form is not submitted, the lack of specific field validation feedback results in poor user experience.

## Reproduction

1. Go to the newsletter section in the footer and enter a input then click on the subscribe button.
2. In the form that appears, try inputting an invalid email address or leaving some inputs blank
3. Click  on submit
4. an error message would appear at the top without  specific validation for the invalid fields.

## Expectation

-  There should be validation messages or features such as(a red border around the input field) in fields with errors, clearly specifying what needs to be corrected.

- The form should highlight the affected fields to help users know how to fix the issue

## Screenshots
Screenshot 2024-10-07 at 07 58 13

## Additional context
Proper form validation with clear feedback on the fields that need correction will improve accessibility and user experience.

## Resolution

- [x] I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by Daodu111**
## Fixes

- Fixes #121 by @possumbilities 

## Description

I added email input validation to ensure users enter a valid email format. Users receive immediate feedback below the input field, with color-coded borders indicating whether their entry is valid (green) or invalid (red). 

## Technical details

1. It uses egular expression to check the email format 
2. it use HTML, CSS and Js

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [ ] My pull request targets the *default* branch of the repository (`main` or `master`).
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

For the purposes of this DCO, "license" is equivalent to "license or public domain dedication," and "open source license" is equivalent to "open content license or public domain dedication."
<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
-.DS_Store
\ No newline at end of file
+.DS_Store
+.vscode
```
Comment (possumbilities) [inline .gitignore]: this change is unrelated to this PR.

Comment (possumbilities) [reply .gitignore]: Also missing End of File

File: specimen/contexts/archive-page.html [hunk 2]
```diff
@@ -294,7 +297,32 @@ <h2>Support Our Work</h2>
...
     </footer>
 
-<script src="../../../src/js/vocabulary.js"></script>
+    <div class="scroll-container">
+        <a href="#top" class="scroll-to-top icon-attach fa-angle-up" aria-label="Scroll to top"></a>
+    </div>
+
+<!--<script src="../../../src/js/vocabulary.js"></script>-->
+    <script>
+        document.addEventListener('DOMContentLoaded', function () {
```
Comment (possumbilities) [inline specimen/contexts/archive-page.html]: Any JS would need to be added to a separate .js file and not embedded within the HTML structure.

Comment (Daodu111) [reply specimen/contexts/archive-page.html]: I would do add it to the Js file now. Also I want to ask if it meets the requirements

File: .gitignore (hunk 1 above)
Comment (possumbilities) [inline .gitignore:2]: There are still changes to this file, this is unrelated to this PR

**Comment by possumbilities** [comment]
Once you've solidified an approach for how to do this, this is a change that would have to be made in the footer of every context to be valid, not just the 'archive-page.html`

File: src/css/vocabulary.css [hunk 3]
```diff
@@ -1407,6 +1407,39 @@ body > footer .subscribe form input.button {
...
+    margin-right: 0px; 
+}
+
+.error-message {
+    font-size: 12px; 
+    color: red; 
+    margin-top: 10px; 
+    display: block; 
+}
+.the-form {
```
Comment (possumbilities) [inline src/css/vocabulary.css:1438]: a class selector here isn't needed. I would suggest something that follows the progression of naming that accounts for specificity, such as: `.subscribe form`.

This would also mean you don't need the class in the HTML markup.

File: src/css/vocabulary.css [hunk 4]
```diff
@@ -1819,6 +1852,50 @@ body > footer .license svg {
     --underline-background-color: var(--vocabulary-brand-color-soft-turquoise);
 }
 
+/* scroll to top */
```
Comment (possumbilities) [inline src/css/vocabulary.css:1855]: There are several changes here related to the scroll-to-top feature that need to be removed.

PRs should be concisely related to the Issue they are addressing only. Please remove this.
